✨ Use cluster name for admin user in kubeconfig (#1698)
kubernetes-sigs/cluster-api: docs/book/src/providers/v1alpha2-to-v1alpha3.md

## The `util/restmapper` package has been removed

- Controller runtime has native support for a [DynamicRESTMapper](https://github.com/kubernetes-sigs/controller-runtime/pull/554/files), which is used by default when creating a new Manager.

> ## Generated kubeconfig secret for clusters now comes with unique per cluster admin username

Reviewer: ```suggestion
## Generated kubeconfig username changed from `admin` to `<cluster-name>-admin`
```

Author: The change was from `kubernetes-admin` to `<ClusterName>-admin`.

> - Default kubeconfig secret now uses cluster name in the admin admin user to ensure unique usernames allowing kubeconfig for multiple clusters to be flatten more easily.

Reviewer: ```suggestion
- The kubeconfig secret shipped with Cluster API now uses the cluster name as prefix to the `username` field.
```
